app/controllers/application_controller.rb

```diff
@@ -3,8 +3,141 @@
 class ApplicationController < Sinatra::Base
 
   configure do
+    set :views, "app/views"
+    enable :sessions
+    set :session_secret, "password_security"
+  end
+
+
+  configure do
     set :public_folder, 'public'
     set :views, 'app/views'
   end
 
-end
\ No newline at end of file
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    if logged_in?
+      redirect '/tweets'
+    end
+    erb :'/users/create_user'
+  end
+
+  post '/signup' do
+    if params[:username] == "" || params[:email] == "" || params[:password] == ""
+      redirect '/signup'
+    else
+      User.create(username: params[:username], email: params[:email], password: params[:password])
+      @user = User.find_by(username: params[:username])
+        session[:user_id] = @user.id
+        redirect to "/tweets"
+    end
+  end
+
+
+
+  get '/login' do
+    if logged_in?
+      redirect '/tweets'
+    else
+      erb :'users/login'
+    end
+  end
+
+  post '/login' do
+    @user = User.find_by(username: params[:username])
+    if @user && @user.authenticate(params[:password])
+      session[:user_id] = @user
+      redirect '/tweets'
+    else
+      redirect '/login'
+    end
+
+  end
+
+  get '/tweets' do
+    if logged_in?
+      @tweets = Tweet.all
+      erb :'/tweets/tweets'
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      erb :'/tweets/create_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+  post '/tweets' do
+    if params[:content] == ""
+      redirect '/tweets/new'
+    else
+      current_user.tweets.create(content: params[:content])
+    end
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'/tweets/show_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'/tweets/edit_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+  delete '/tweets/:id' do
+    @tweet = Tweet.find_by_id(params[:id])
+    @tweet.delete unless current_user.id != @tweet.user_id
+    redirect '/tweets'
+  end
+
+  patch '/tweets/:id' do
+    @tweet = Tweet.find_by_id(params[:id])
+
+    if params[:content] == ""
+      redirect "/tweets/#{@tweet.id}/edit"
+    else
+      @tweet.content = params[:content]
+      @tweet.save
```

Comment:
I would contend that you also want to double check the user's association to the tweet here. There are other ways to submit a PATCH/POST request than going through an HTML form...